diff -r -u -x configure dams-4.3.4-orig/lib/DamsWrapper.cpp dams-4.3.4-dev/lib/DamsWrapper.cpp
--- dams-4.3.4-orig/lib/DamsWrapper.cpp	2015-12-25 08:03:59.000000000 +0900
+++ dams-4.3.4-dev/lib/DamsWrapper.cpp	2020-05-18 21:29:39.320445296 +0900
@@ -187,7 +187,10 @@
       score = 0;
     } else {
       score = result.score;
-      std::string euc_tail = std::string((char*)(euc_query.c_str() + matchlen));
+      std::string euc_tail("");
+      if (euc_query.length() > matchlen){
+          euc_tail = euc_query.substr(matchlen);
+      }
       iconv_to_utf8->convert(euc_tail);
       tail = iconv_to_utf8->getConverted();
       for (int i = 0; i < result.res.Elements() && i < limits; i++) {
